warehouse/packaging/models.py

```diff
@@ -268,6 +270,37 @@ def latest_version(self):
             .first()
         )
 
+    @property
+    def latest_stable_version(self):
```

Comment:
What if those properties returned a `Release` object instead of just a few fields ? I believe it would simplify the View code.

Reply:
Huh. Yeah, it looks like the test suite still passes fine if I switch the three `latest` properties to just query into `Release`:

```
    @property
    def latest_version(self):
        # Supply the latest stable version, if any stable releases exist.
        # If only pre-releases exist, supply the latest pre-release version.
        return (
            orm.object_session(self)
            .query(Release)                   <--------
            .filter(...
```

So, yeah, that should enable the tightening of the view code as you suggest. I'll make those changes.